src/core/Form/TextInput/TextInput.baseStyles.tsx

```diff
@@ -1,11 +1,13 @@
 import { css } from 'styled-components';
 import { withSuomifiTheme, TokensAndTheme } from '../../theme';
-import { input, inputContainer } from '../../theme/reset';
+import { input, inputContainer, font } from '../../theme/reset';
 
 export const baseStyles = withSuomifiTheme(
   ({ theme }: TokensAndTheme) => css`
   & .fi-text-input_label-p {
-    margin-bottom: ${theme.spacing.insetXl};
+    margin-bottom: ${theme.spacing.insetL};
+    ${font({ theme })('actionElementInnerTextBold')};
+    color: ${theme.colors.blackBase};
```

Comment:
This was not mentioned in the description but seems to fix the issue when parent container's `color` was changing the color of the label.

Reply:
Counted this as a style fix, so didn't mention it separately. Maybe should've. But yeah, it should be fixed by this.